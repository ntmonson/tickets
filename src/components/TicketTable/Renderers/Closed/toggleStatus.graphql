mutation toggleTicketStatus($id: String!) {
  databaseToggleTicketStatus(id: $id) {
    id
    topic
    closed
    pinned
    createdAt
    updatedAt
  }
}
