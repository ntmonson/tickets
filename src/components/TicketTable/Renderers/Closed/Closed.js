import React from 'react';
import PropTypes from 'prop-types';
import { graphql, compose } from 'react-apollo';
import toggleStatus from './toggleStatus.graphql';

// Exported for testing, see https://github.com/kriasoft/react-starter-kit/issues/378
export const Closed = ({ value: closed, ticketId, toggleStat }) => {
  const color = closed ? 'Red' : '#373277';
  return (
    <span>
      {closed ? 'Closed' : 'Open'}
      <button style={{ color }} onClick={() => toggleStat(ticketId)}>
        {closed ? (
          <i className="fas fa-toggle-off" />
        ) : (
          <i className="fas fa-toggle-on" />
        )}
      </button>
    </span>
  );
};

Closed.propTypes = {
  value: PropTypes.number.isRequired,
  ticketId: PropTypes.string.isRequired,
  toggleStat: PropTypes.func.isRequired,
};

const graphqlQueries = graphql(toggleStatus, {
  name: 'toggleStat',
  props: ({ toggleStat }) => ({
    toggleStat: id => toggleStat({ variables: { id } }),
  }),
});

export default compose(graphqlQueries)(Closed);
