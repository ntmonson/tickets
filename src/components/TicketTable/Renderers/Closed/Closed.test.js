/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import { Closed } from './Closed';

describe('Closed', () => {
  test('Closed renders correctly with value 1', () => {
    const component = renderer
      .create(<Closed value={1} ticketId="ASDF1234" toggleStat={() => {}} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('Closed renders correctly with value 0', () => {
    const component = renderer
      .create(<Closed value={0} ticketId="ASDF1234" toggleStat={() => {}} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
