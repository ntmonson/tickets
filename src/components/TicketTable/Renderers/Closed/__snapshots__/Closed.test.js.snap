// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Closed Closed renders correctly with value 0 1`] = `
<span>
  Open
  <button
    onClick={[Function]}
    style={
      Object {
        "color": "#373277",
      }
    }
  >
    <i
      className="fas fa-toggle-on"
    />
  </button>
</span>
`;

exports[`Closed Closed renders correctly with value 1 1`] = `
<span>
  Closed
  <button
    onClick={[Function]}
    style={
      Object {
        "color": "Red",
      }
    }
  >
    <i
      className="fas fa-toggle-off"
    />
  </button>
</span>
`;
